src/Substance.hs

```diff
@@ -4,535 +4,450 @@
-
--- 'checkReferencess' checks any statement that refers to other objects. For example,
--- > Subset A B
--- refers to identifiers @A@ and @B@. The function will perform a lookup in the symbol table, make sure the objects exist and are of desired types -- in this case, 'Set' -- and throws an error otherwise.
-checkReferencess :: SubEnv -> SubStmt -> SubEnv
-checkReferencess e (ConstrDecl t ss)  = e { subObjs = newConstrs : subObjs e }
-    where newConstrs = toConstr t $ map (checkNameAndTyp $ subSymbols e) $ zip ss ts
-          ts = case t of
-              PointInT    -> [PointT, SetT]
-              PointNotInT -> [PointT, SetT]
-              _           -> [SetT, SetT]
-checkReferencess e (FuncVal f a b)  = e { subObjs = val : subObjs e }
-    where args = map (checkNameAndTyp $ subSymbols e) $ zip [f, a, b] [MapT, PointT, PointT]
-          val  = toObj ValueT args
-checkReferencess e (MapInit t f a b) = e { subObjs = toObj t args : subObjs e }
-    where args = map (checkNameAndTyp $ subSymbols e) $ zip [f, a, b] [MapT, SetT, SetT]
-checkReferencess e (DefApp n args) = e { subApps = (def, apps, funcToSolve ++ setsToSolve) : subApps e }
-    where (sigs, def)  = fromMaybe (error ("Definition " ++ n ++ " does not exist.")) (M.lookup n (subDefs e))
-          argsWithTyps = zip args $ map fst sigs
-          args' = map (checkNameAndTyp $ subSymbols e) argsWithTyps
-          apps  = M.fromList $ zip (map snd sigs) args
-          funcToSolve = map fst $ filter (\(n, t) -> t == MapT) argsWithTyps
-          setsToSolve =  concatMap (\x -> fromMaybe (error ("Function " ++ x ++ "does not exist.")) (M.lookup x $ subArgs e)) funcToSolve
-checkReferencess e _ = e -- Ignore all other statements
+-- | 'passDecls' checks the validity of declarations of objects.
+passDecls :: VarEnv -> SubObjects -> SubStmt -> SubObjects
+passDecls subEnv e (Decl t s)  = e { subObjs = toObj subEnv t s : subObjs e }
+passDecls subEnv e _ = e -- Ignore all other statements
+
+-- | 'toObj' translates [Type] + [Identiers] to concrete Substance objects, to be selected by the Style program
+toObj :: VarEnv -> T -> Var -> SubObj
+toObj e t v = LD $ (SubDeclConst (fixAST e t) v)
+
+fixAST :: VarEnv -> T -> T
+fixAST e (TConstr c) = TConstr (c { argCons = (map (fixArg e) (argCons c)) })
+fixAST e t = t -- Ignore all other cases
+
+fixArg :: VarEnv -> Arg -> Arg
+fixArg e (AT (TConstr i)) = if (nameCons i) `elem` (declaredNames e)
+                            then AVar (VarConst (nameCons i))
+                            else AT (TConstr i)
+fixArg e a = a -- Ignore all other cases
+
+-- | 'checkReferencess' checks any statement that refers to other objects. For example,
+-- | > Subset A B
+-- | refers to identifiers @A@ and @B@. The function will perform a lookup in the symbol table, make sure the objects exist and are of desired types -- in this case, 'Set' -- and throws an error otherwise.
+passReferencess :: SubObjects -> SubStmt -> SubObjects
```

Comment:
needs a more accurate name and comment after the refactor

Reply:
Again might change after style is implemented